packages/rich-text/src/plugins/Tracking/createTrackingPlugin.ts

```diff
@@ -0,0 +1,45 @@
+import * as contentfulSlateJSAdapter from '@contentful/contentful-slatejs-adapter';
+import { documentToPlainTextString } from '@contentful/rich-text-plain-text-renderer';
+import { PlateEditor } from '@udecode/plate-core';
+
+import Schema from '../../constants/Schema';
+import { TrackingProvider } from '../../TrackingProvider';
+import { RichTextPlugin } from '../../types';
+
+function getCharacterCount(editor: PlateEditor) {
+  const document = contentfulSlateJSAdapter.toContentfulDocument({
+    document: editor.children,
+    schema: Schema,
+  });
+
+  return documentToPlainTextString(document).length;
+}
+
+export const createTrackingPlugin = (tracking: TrackingProvider): RichTextPlugin => ({
```

Comment:
Since we have a plugin anyway, how do you feel about getting rid of the `tracking` value we pass everywhere (+ the `TrackingProvider` ofc) and instead inject `tracking` property in the `editor` instance?

* Lots of plugins already do something similar so I don't think it's a bad practice
* We already have access to the `editor` instance everywhere in the codebase, whether it's a React component or a pure JS helper.

Something like this:

```javascript
editor.tracking?.logShortcutAction(...)
```